### PR TITLE
feat: support inverse embeds relationship

### PR DESCRIPTION
**Pull request by willtj**
To get from an embedded sub document back to its parent, eg to support `touchOwners()` updating the parent timestamps on child save.

### CODE REVIEW

File: src/Jenssegers/Mongodb/Eloquent/EmbedsRelations.php [hunk 1]
```diff
@@ -44,7 +45,7 @@ protected function embedsMany($related, $localKey = null, $foreignKey = null, $r
     }
 
     /**
-     * Define an embedded one-to-many relationship.
+     * Define an embedded one-to-one relationship.
```
Comment (StevePorter92) [inline src/Jenssegers/Mongodb/Eloquent/EmbedsRelations.php:48]: CONTRIBUTION POINTS

File: src/Jenssegers/Mongodb/Eloquent/EmbedsRelations.php [hunk 2]
```diff
@@ -77,4 +78,33 @@ protected function embedsOne($related, $localKey = null, $foreignKey = null, $re
...
+     * @param  string $localKey
+     * @param  string $foreignKey
+     * @param  string $relation
+     * @return \Jenssegers\Mongodb\Relations\EmbeddedBy
+     */
+    protected function embeddedBy($related, $foreignKey = null, $ownerKey = null, $relation = null)
+    {
+        // Use debug backtrace to extract the calling method's name and use that as
+        // the relationship name
+        list(, $caller) = debug_backtrace(false);
```
Comment (StevePorter92) [inline src/Jenssegers/Mongodb/Eloquent/EmbedsRelations.php:95]: i dont understand this, but this looks cool

File: src/Jenssegers/Mongodb/Relations/EmbeddedBy.php [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    /**
+     * Set the constraints for an eager load of the relation.
+     *
+     * @param  array  $models
+     * @return void
+     */
+    public function addEagerConstraints(array $models)
+    {
+
```
Comment (StevePorter92) [inline src/Jenssegers/Mongodb/Relations/EmbeddedBy.php:30]: throw an exception here?

**Review by StevePorter92 — APPROVED** [review]
Really nice

File: src/Jenssegers/Mongodb/Eloquent/EmbedsRelations.php (hunk 2 above)
Comment (willtj) [reply src/Jenssegers/Mongodb/Eloquent/EmbedsRelations.php:95]: 🤷‍♂️

File: src/Jenssegers/Mongodb/Relations/EmbeddedBy.php (hunk 3 above)
Comment (willtj) [reply src/Jenssegers/Mongodb/Relations/EmbeddedBy.php:30]: Looks like Jensy leaves this method blank too so I think it's ok as is. I was about to add exceptions to methods that aren't implemented but realised I don't actually know what's already working (and required) as it is, so I think we might just have to be aware that there could be strange issues around sub documents if we try to do any more than we're currently using them for.

Comment (StevePorter92) [reply src/Jenssegers/Mongodb/Relations/EmbeddedBy.php:30]: Good thinking.

**Pull request merged**